src/client/common/prompters.cpp

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (C) 2021 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <multipass/cli/prompters.h>
+#include <multipass/exceptions/cmd_exceptions.h>
+
+#include <iostream>
+
+namespace multipass
+{
+
+std::string PlainPrompter::prompt(const std::string& text)
+{
+    term->cout() << text << ": ";
+
+    std::string value;
+    std::getline(term->cin(), value);
+
+    if (term->cin().eof())
```

Comment:
Oh one more thing, I think this should check for `fail()` too/instead. The [table at the bottom here](https://en.cppreference.com/w/cpp/io/basic_ios/fail) is helpful. I am not sure if we want `eof()` but `!fail()` to be a valid way to express the empty string (e.g. when redirecting stdin).

Reply:
Indeed! I'm on the fence about `eof()`, though. This is meant to be interactive, so is redirection a use case we should optimize for? And if not, is EOF expected when interactive? If you close your terminal at the prompt, should that count as empty string input?

Reply:
I've gone for rejecting `!good()`.

Reply:
Makes sense. Thanks!